Sources/Logging/Logging.swift

```diff
@@ -96,25 +96,46 @@ public struct Logger {
     }
 }
 
-// This is the logging system itself, it's mostly used to obtain loggers and to set the type of the `LogHandler`
-// implementation.
-public enum Logging {
-    private static let lock = ReadWriteLock()
-    private static var _factory: (String) -> LogHandler = StdoutLogger.init
+private let logHandlerFactoryLock = ReadWriteLock()
+private var logHandlerFactory: (String) -> LogHandler = StdoutLogHandler.init
+private let globalLabel: String = {
+    // Taken from Foundation.UUID
+    var uuid: uuid_t = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+    withUnsafeMutablePointer(to: &uuid) {
+        $0.withMemoryRebound(to: UInt8.self, capacity: MemoryLayout<uuid_t>.size) {
+            uuid_generate_random($0)
+        }
+    }
+    var bytes: uuid_string_t = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+    return withUnsafePointer(to: uuid) { valPtr in
+        valPtr.withMemoryRebound(to: UInt8.self, capacity: MemoryLayout<uuid_t>.size) { val in
+            withUnsafeMutablePointer(to: &bytes) { strPtr in
+                strPtr.withMemoryRebound(to: CChar.self, capacity: MemoryLayout<uuid_string_t>.size) { str in
+                    uuid_unparse_lower(val, str)
+                    return String.init(cString: str)
+                }
+            }
+        }
+    }
+}()
 
-    // Configures which `LogHandler` to use in the application.
+extension Logger {
     public static func bootstrap(_ factory: @escaping (String) -> LogHandler) {
-        self.lock.withWriterLock {
-            self._factory = factory
+        logHandlerFactoryLock.withWriterLock {
+            logHandlerFactory = factory
         }
     }
 
-    public static func make(_ label: String) -> Logger {
-        return self.lock.withReaderLock { Logger(self._factory(label)) }
+    public init(label: String) {
+        self = logHandlerFactoryLock.withReaderLock { Logger(logHandlerFactory(label)) }
+    }
+
+    public init() {
```

Comment:
can we remove the initialiser without label?

Reply:
Do you think the convenience dangerously pushes developers towards the one-global logger?

Reply:
I think it’s quite useful to have the logger at least identify themselves with some form of ID string. And I think the burden to the user passing a string there is not too high. But offering one without the need of passing an ID string is just too easy so I fear that no one will end up using the ID strings and they quickly become useless.

Sure, the users could just pass the empty string etc but I don’t think they would.

It’s basically like `DispatchQueue(label: “foo.bar”)` and people seem to fill in more or less sensible stuff.

Reply:
Makes sense. Will make the necessary changes